From 7d1e22e3af88b8620cf806c881aa4e46266465c0 Mon Sep 17 00:00:00 2001
From: Pardha Kesapragada <pkesapragada@habana.ai>
Date: Wed, 17 Feb 2021 19:06:35 +0200
Subject: [PATCH] [SW-20154] add function to set spi write limit

SPI write limit for Gaudi2 is just 2MB. Hence make
override this at a platform level

Change-Id: I1c0db67582a0918735c4975b854b3488d41ec5f9
Signed-off-by: Pardha Kesapragada <pkesapragada@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index fccfe3c..52df15a 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -115,6 +115,11 @@
 	security_disable_jtag_if();
 }
 
+__attribute__((weak)) void arcboot_rom_set_spi_write_limit(void)
+{
+	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
+}
+
 static void arcboot_rom_init(void)
 {
 	uint32_t reg;
@@ -148,7 +153,9 @@
 		 COLD_RST_FLOPS_EFUSE_READ_SHIFT;
 
 	regmap_writel(SPI_FLASH_BAUDR_DIVISOR, mmPSOC_QSPI_BAUDR);
-	regmap_writel(write_en_16MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
+
+	arcboot_rom_set_spi_write_limit();
+
 	if (fuse_init() == 0)
 		arcboot_rom_set_efuse_read();
 
diff --git a/gaudi2_arc_gnu/common.c b/gaudi2_arc_gnu/common.c
index 5e48eaf..1703849 100644
--- a/gaudi2_arc_gnu/common.c
+++ b/gaudi2_arc_gnu/common.c
@@ -79,3 +79,9 @@
 	/* Further, load the recovery bin from eROM */
 
 }
+
+void arcboot_rom_set_spi_write_limit(void)
+{
+	/* limit flash write to just 2MB */
+	regmap_writel(write_en_2MB, mmPSOC_SECURITY_SPI_WRITE_LIMIT);
+}
